File: src/main/java/org/jukeboxmc/block/behavior/BlockCoralFanDead.java
package org.jukeboxmc.block.behavior;

import com.nukkitx.nbt.NbtMap;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.block.Block;
import org.jukeboxmc.block.data.CoralColor;
import org.jukeboxmc.block.direction.RotationDirection;
import org.jukeboxmc.item.Item;
import org.jukeboxmc.item.ItemType;
import org.jukeboxmc.item.behavior.ItemCoralFanDead;
import org.jukeboxmc.util.Identifier;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockCoralFanDead extends Block {

    public BlockCoralFanDead( Identifier identifier ) {
        super( identifier );
    }

    public BlockCoralFanDead( Identifier identifier, NbtMap blockStates ) {
        super( identifier, blockStates );
    }

    @Override
    public Item toItem() {
        return Item.<ItemCoralFanDead>create( ItemType.CORAL_FAN_DEAD ).setCoralColor( this.getCoralColor() );
    }

    public void setCoralDirection(@NotNull RotationDirection rotationDirection ) {
        this.setState( "coral_fan_direction", rotationDirection.ordinal() );
    }

    public RotationDirection getRotationDirection() {
        return this.stateExists( "coral_fan_direction" ) ? RotationDirection.values()[this.getIntState( "coral_fan_direction" )] : RotationDirection.EAST_WEST;
    }

    public BlockCoralFanDead setCoralColor(@NotNull CoralColor coralColor ) {
        return this.setState( "coral_color", coralColor.name().toLowerCase() );
    }

    public @NotNull CoralColor getCoralColor() {
        return this.stateExists( "coral_color" ) ? CoralColor.valueOf( this.getStringState( "coral_color" ) ) : CoralColor.BLUE;
    }
}

File: src/main/java/org/jukeboxmc/block/behavior/BlockCoralFanDead.kt
package org.jukeboxmc.block.behavior

import com.nukkitx.nbt.NbtMap
import org.jukeboxmc.block.Block
import org.jukeboxmc.block.data.CoralColor
import org.jukeboxmc.block.direction.RotationDirection
import org.jukeboxmc.item.Item
import org.jukeboxmc.item.ItemType
import org.jukeboxmc.item.behavior.ItemCoralFanDead
import org.jukeboxmc.util.Identifier
import java.util.Locale

/**
 * @author LucGamesYT
 * @version 1.0
 */
class BlockCoralFanDead : Block {
    constructor(identifier: Identifier?) : super(identifier)
    constructor(identifier: Identifier?, blockStates: NbtMap?) : super(identifier, blockStates)

    override fun toItem(): Item {
        return Item.create<ItemCoralFanDead>(ItemType.CORAL_FAN_DEAD).setCoralColor(
            coralColor,
        )
    }

    fun setCoralDirection(rotationDirection: RotationDirection) {
        setState<Block>("coral_fan_direction", rotationDirection.ordinal)
    }

    val rotationDirection: RotationDirection
        get() = if (stateExists("coral_fan_direction")) RotationDirection.values()[getIntState("coral_fan_direction")] else RotationDirection.EAST_WEST

    fun setCoralColor(coralColor: CoralColor): BlockCoralFanDead {
        return setState("coral_color", coralColor.name.lowercase(Locale.getDefault()))
    }

    val coralColor: CoralColor
        get() = if (stateExists("coral_color")) CoralColor.valueOf(getStringState("coral_color")) else CoralColor.BLUE
}
